### PR TITLE
documentation: adds contributing.md

### PR DESCRIPTION
**Pull request by priley86**
Adds code contribution and contribution process guidelines for the Patternfly React project.

issue: #4 

cc: @jgiardino @maryclarke @LHinson @bleathem

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
+# Contributing to Patternfly React
```
Comment (LHinson) [inline CONTRIBUTING.md]: all instances of "Patternfly" should be updated to "PatternFly"

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+- `pending response` - This label indicates that the issue is awaiting reply from a contributor or a maintainer.
+- `question` - Use the question label to indicate you are having a problem with something and would like feedback from the community.
+- `wontfix` - The issue is legitimate, but it is not something the team is currently able or willing to fix or implement. Issues with this label may be revisited in the future.
+
+## React Component Contribution Process
+
+React component contributions should observe the following high level process:
+
+1. Check the [Patternfly Decision Tree](https://github.com/patternfly/patternfly-design/blob/master/resources/decision-tree/PatternflyDecisionTree.pdf) to see if your component meets the criteria for an addition to Patternfly.
+2. Create an `enhancement` issue to indicate you'd like to contribute this component to Patternfly React and notify the Patternfly React maintainer team.
```
Comment (LHinson) [inline CONTRIBUTING.md]: you reference the "PatternFly React maintainer team" however it does not provide who this anywhere. As someone reading this and unfamiliar with the team, I wouldn't be sure who to reach out to. What do you think about identifying the team members?

Comment (LHinson) [inline CONTRIBUTING.md]: "Create an `enhancement` issue to indicate you'd like to contribute this component to Patternfly React and notify the Patternfly React maintainer team." <<< via? how does the team want contacted? the mailing list?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+The [issue tracker](https://github.com/patternfly/patternfly-react/issues) is the preferred channel for bug reports, features requests, and submitting pull requests.
+
+### Issues and labels
+
+Our issue tracker utilizes several labels to help organize and identify issues. Here's what they represent and how we use them:
+
+- `bug` - A bug is a _demonstrable problem_ that is caused by the code in the repository. Please check if the issue has already been reported before reporting a new bug.
+- `duplicate` - A duplicate signifies that another issue encapsulates or restates this issue. Duplicates will be closed and referenced as such. 
+- `ehancement` - An ehancement is a feature request. Please provide as much detail as possible and consider whether your idea fits the scope and aims of this project.
```
Comment (jgiardino) [inline CONTRIBUTING.md]: Typo here for 'enhancement'

File: CONTRIBUTING.md (hunk 2 above)
Comment (jgiardino) [inline CONTRIBUTING.md]: If we're all subscribed to this repo, then will we automatically get notifications when new issues are created? If so, then I ok with just saying "Create an `enhancement` issue to indicate you'd like to contribute this component to PatternFly React."

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+when possible and accept [props](https://facebook.github.io/react/docs/components-and-props.html) as UI display parameters.
+* Provide a [single default export](http://exploringjs.com/es6/ch_modules.html#_single-default-export) for exporting your React UI component as an ES6 Module in your component's jsx definition.
+* Provide an associated `.stories.js` [story](https://getstorybook.io/docs/react-storybook/basics/writing-stories) for your component. Stories should demonstrate as many different UI states for your component as possible. Use Storybook [knobs](https://github.com/storybooks/storybook-addon-knobs) to enable dynamic visualizations of your component's props.
+* Provide a [jest snapshot test](https://facebook.github.io/jest/docs/snapshot-testing.html) to ensure your UI markup does not change unexpectedly.
+* Ensure the component's rendered design and Storybook stories meet [Patternfly design standard](https://github.com/patternfly/patternfly-design). 
+**Note:** If your component does not yet have Patternfly design documentation, please meet with the Patternfly React maintainer team to ensure this gets contributed along with your component.
```
Comment (jgiardino) [inline CONTRIBUTING.md]: Similar to Leslie's comment above, this states to contact team members but doesn't describe the process for that. Any suggestions/preferences on how the design team wants to handle this process?

If the necessary reviewers will already receive notifications, we could just change this statement to something like "If your component does not yet have Patternfly design documentation, the PatternFly React design team will first confirm that the pattern passes the [Patternfly Decision Tree](https://github.com/patternfly/patternfly-design/blob/master/resources/decision-tree/PatternflyDecisionTree.pdf) and then start the process for generating design documentation"

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+
+6. Push
+
+  ```text
+  $ git push origin my-branch
+  ```
+
+7. Export Storybook 
+
+Lastly, you'll want to export Storybook in your fork and note the url afterward.
```
Comment (jgiardino) [inline CONTRIBUTING.md]: can we change this to 
`...and include the Storybook url in your PR.`

Comment (jgiardino) [inline CONTRIBUTING.md]: Oh, I see you have that below :-)

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+
+## React Component Requirements 
+
+Please ensure that all React UI components contributed meet the following guidelines:
+
+* This repository serves as a UI / presentational component library only. This means we should not be introducing container components which subscribe to state updates or handle data fetching (i.e. redux aware components). Prefer [stateless functional components](http://buildwithreact.com/article/stateless-functional-components)
+when possible and accept [props](https://facebook.github.io/react/docs/components-and-props.html) as UI display parameters.
+* Provide a [single default export](http://exploringjs.com/es6/ch_modules.html#_single-default-export) for exporting your React UI component as an ES6 Module in your component's jsx definition.
+* Provide an associated `.stories.js` [story](https://getstorybook.io/docs/react-storybook/basics/writing-stories) for your component. Stories should demonstrate as many different UI states for your component as possible. Use Storybook [knobs](https://github.com/storybooks/storybook-addon-knobs) to enable dynamic visualizations of your component's props.
```
Comment (jgiardino) [inline CONTRIBUTING.md]: Should we also include comments about the template that we have defined, such as: 

> Include a link to the design documentation in the [PatternFly Pattern Library[(http://www.patternfly.org/pattern-library/) if it exists. If a PatternFly design does not exist yet, then provide a description that explains when the component would be used and what goal or task it helps to accomplish.

If we do add this info, then maybe we should move some of the storybook details to sub-bullets?

**Comment by jgiardino** [comment]
Thanks for doing this, @priley86! I included some comments inline. Do we also want to state somewhere in this document that if someone contributes to this repo, that we would also like them to participate in reviewing PRs from other contributors?

File: CONTRIBUTING.md (hunk 1 above)
Comment (priley86) [inline CONTRIBUTING.md]: will update

File: CONTRIBUTING.md (hunk 3 above)
Comment (priley86) [inline CONTRIBUTING.md]: good catch. will fix.

File: CONTRIBUTING.md (hunk 6 above)
Comment (priley86) [inline CONTRIBUTING.md]: Sounds good to me @jgiardino. Will update.

**Comment by priley86** [comment]
Can you provide some language for this @jgiardino? Happy to provide that suggestion...

File: CONTRIBUTING.md (hunk 2 above)
Comment (LHinson) [inline CONTRIBUTING.md]: That approach makes sense to me @jgiardino

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+## React Component Contribution Process
+
+React component contributions should observe the following high level process:
+
+1. Check the [Patternfly Decision Tree](https://github.com/patternfly/patternfly-design/blob/master/resources/decision-tree/PatternflyDecisionTree.pdf) to see if your component meets the criteria for an addition to Patternfly.
+2. Create an `enhancement` issue to indicate you'd like to contribute this component to Patternfly React and notify the Patternfly React maintainer team.
+3. Implement the component observing our [react component requirements](#react-component-requirements) 
+and [code contribution guidelines](#code-contribution-guidelines)
+4. Submit your pull request and be sure to reference the `enhancement` issue so that others interested in this component may discuss. 
+5. Test your component. Once the pull request is merged, please test the component can successfully be used as intended downstream.
```
Comment (jgiardino) [inline CONTRIBUTING.md]: As a final statement (i.e. not a list item), lets also include:

> If you submit a PR, then please also contribute by reviewing other PRs.

**Comment by jgiardino** [comment]
Included a suggested statement for reviewing PRs in this section:
## React Component Contribution Process

File: CONTRIBUTING.md (hunk 2 above)
Comment (jgiardino) [inline CONTRIBUTING.md]: Updating my suggestion to include more details about the review process:

> Create an enhancement issue to indicate you'd like to contribute this component to PatternFly React. At this time, the PatternFly React design team will review the enhancement issue and verify that the component passes the decision tree. Only components that pass will be accepted into the repo. If the component does not pass, the team will work with you on identifying any sub-components that can be extracted that would be needed by more than one project.

Please review and let me know if this accurately captures what @maryclarke and @wrnkld and Rox will do. I don't want to make any promises. :-) 

Also let me know if this should be broken out somewhere else.

Comment (jgiardino) [inline CONTRIBUTING.md]: @LHinson Do other repo teams state who their core members are? If so, where do they add this information?

**Comment by priley86** [comment]
I've updated this with a note about `semantic-releases` should we decide to go with this option (see proposal in #13. Also added a Github `PULL_REQUEST_TEMPLATE`. 

The Travis CI build is fixed in #13 and can be ignored in this PR.

Let me know anymore thoughts here before we take flight! 🚀

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+- `help wanted` - Issues we need or would love help from the community to resolve.
+- `pending response` - This label indicates that the issue is awaiting reply from a contributor or a maintainer.
+- `question` - Use the question label to indicate you are having a problem with something and would like feedback from the community.
+- `wontfix` - The issue is legitimate, but it is not something the team is currently able or willing to fix or implement. Issues with this label may be revisited in the future.
+
+## React Component Contribution Process
+
+React component contributions should observe the following high level process:
+
+1. Check the [PatternFly Decision Tree](https://github.com/patternfly/patternfly-design/blob/master/resources/decision-tree/PatternflyDecisionTree.pdf) to see if your component meets the criteria for an addition to PatternFly.
```
Comment (priley86) [inline CONTRIBUTING.md]: +1 will add it.

Comment (priley86) [inline CONTRIBUTING.md]: Added :)

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+2. Create an `enhancement` issue to indicate you'd like to contribute this component to PatternFly React. At this time, the PatternFly React design team will review the enhancement issue and verify that the component passes the decision tree. Only components that pass will be accepted into the repo. If the component does not pass, the team will work with you on identifying any sub-components that can be extracted that would be needed by more than one project.
+3. Implement the component observing our [react component requirements](#react-component-requirements)
```
Comment (priley86) [inline CONTRIBUTING.md]: Added :)

**Comment by priley86** [comment]
needs rebase ;)

**Comment by LHinson** [comment]
Looks good to merge and we can continue to iterate on it! @priley86

**Pull request merged**